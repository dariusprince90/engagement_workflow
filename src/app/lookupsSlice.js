import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import clientsApi from '../api/centralDataStore/clientsApi';
import jobsApi from '../api/centralDataStore/jobsApi';

// **********************************************************************
// * constants

/**
 * The name of the slice (to be used in the slice and thunks).
 */
export const SLICE_NAME = 'lookups';

// **********************************************************************
// * functions

// **********************************************************************
// * thunks

/**
 * Ensure a client exists in the cache for a given client id.
 */
export const ensureClientExistsInCache = createAsyncThunk(
  'lookups/ensureClientExistsInCache',
  async (clientId, { getState, dispatch, rejectWithValue }) => {
    const state = getState();
    const fields = 'id,displayName';

    // if the clientId is already in the pending load collection or in the load error collection
    // then, do nothing (we don't need to attempt to load the same client multiple times)
    if (
      state.lookups.clientsCache.clientIdsPendingLoad.includes(clientId) ||
      state.lookups.clientsCache.clientIdsWithLoadError.includes(clientId)
    ) {
      return;
    }

    // trigger the start of the load
    dispatch(clientLoadingStarted(clientId));

    // try to get the client from the cache
    let client = state.lookups.clientsCache.clients.find((j) => j.id === clientId);

    // if the client was not in the cache
    // then, load the client from the API and insert it into the cache
    if (!client) {
      client = await clientsApi.getClient(clientId, fields);

      // if the client was not returned from the API
      // then there is an error, return a rejected response
      if (!client) {
        // trigger the end of the load
        dispatch(clientLoadingCompleted(clientId));

        // reject
        return rejectWithValue();
      }

      // add the client to the cache
      dispatch(clientAddedToCache(client));
    }

    // trigger the end of the load
    dispatch(clientLoadingCompleted(clientId));
  }
);

/**
 * Ensure a job exists in the cache for a given job id.
 */
export const ensureJobExistsInCache = createAsyncThunk(
  'lookups/ensureJobExistsInCache',
  async (jobId, { getState, dispatch, rejectWithValue }) => {
    const state = getState();
    const fields = 'id,externalProjectReference,clientId';

    // if the jobId is already in the pending load collection or in the load error collection
    // then, do nothing (we don't need to attempt to load the same job multiple times)
    if (
      state.lookups.jobsCache.jobIdsPendingLoad.includes(jobId) ||
      state.lookups.jobsCache.jobIdsWithLoadError.includes(jobId)
    ) {
      return;
    }

    // trigger the start of the load
    dispatch(jobLoadingStarted(jobId));

    // try to get the job from the cache
    let job = state.lookups.jobsCache.jobs.find((j) => j.id === jobId);

    // if the job was not in the cache
    // then, load the job from the API and insert it into the cache
    if (!job) {
      job = await jobsApi.getJob(jobId, fields);

      // if the job was not returned from the API
      // then there is an error, return a rejected response
      if (!job) {
        // trigger the end of the load
        dispatch(jobLoadingCompleted(jobId));

        // reject
        return rejectWithValue();
      }

      // add the job to the cache
      dispatch(jobAddedToCache(job));
    }

    // trigger the end of the load
    dispatch(jobLoadingCompleted(jobId));
  }
);

// **********************************************************************
// * initial state

const initialState = {
  // cache of client objects that can be re-used without having to hit the API each time we need the client
  clientsCache: {
    // the collection of clients that have been cached
    clients: [],

    // collection of client ids that are pending load from the API
    clientIdsPendingLoad: [],

    // collection of client ids that had an error when loading
    clientIdsWithLoadError: []
  },

  // cache of job objects that can be re-used without having to hit the API each time we need the job
  jobsCache: {
    // the collection of jobs that have been cached
    jobs: [],

    // collection of job ids that are pending load from the API
    jobIdsPendingLoad: [],

    // collection of job ids that had an error when loading
    jobIdsWithLoadError: []
  }
};

// **********************************************************************
// * slice

const lookupsSlice = createSlice({
  name: 'lookups',
  initialState,

  reducers: {
    /**
     * Adds the client to the cache.
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    clientAddedToCache: (state, action) => {
      const client = action.payload;
      state.clientsCache.clients.push(client);
    },

    /**
     * Removes the client id from the pending load collection.
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    clientLoadingCompleted: (state, action) => {
      const clientIdToRemove = action.payload;

      state.clientsCache.clientIdsPendingLoad = state.clientsCache.clientIdsPendingLoad.filter(
        (clientId) => clientId !== clientIdToRemove
      );
    },

    /**
     * Adds the client id to the pending load collection (if not already in it).
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    clientLoadingStarted: (state, action) => {
      const clientId = action.payload;

      if (!state.clientsCache.clientIdsPendingLoad.includes(clientId)) {
        state.clientsCache.clientIdsPendingLoad.push(clientId);
      }
    },

    /**
     * Adds the job to the cache.
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    jobAddedToCache: (state, action) => {
      const job = action.payload;
      state.jobsCache.jobs.push(job);
    },

    /**
     * Removes the job id from the pending load collection.
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    jobLoadingCompleted: (state, action) => {
      const jobIdToRemove = action.payload;

      state.jobsCache.jobIdsPendingLoad = state.jobsCache.jobIdsPendingLoad.filter((jobId) => jobId !== jobIdToRemove);
    },

    /**
     * Adds the job id to the pending load collection (if not already in it).
     * @param {object} state - The redux state object.
     * @param {object} action - The action that was dispatched.
     */
    jobLoadingStarted: (state, action) => {
      const jobId = action.payload;

      if (!state.jobsCache.jobIdsPendingLoad.includes(jobId)) {
        state.jobsCache.jobIdsPendingLoad.push(jobId);
      }
    }
  },

  extraReducers: (builder) => {
    builder
      // **********************************************************************
      // * ensureClientExistsInCache
      .addCase(ensureClientExistsInCache.rejected, (state, action) => {
        const clientId = action.meta.arg;

        // if the client id is not in the error collection, add it
        if (!state.clientsCache.clientIdsWithLoadError.includes(clientId)) {
          state.clientsCache.clientIdsWithLoadError.push(clientId);
        }

        // if the client id is in the pending load collection, remove it
        if (state.clientsCache.clientIdsPendingLoad.includes(clientId)) {
          state.clientsCache.clientIdsPendingLoad = state.clientsCache.clientIdsPendingLoad.filter(
            (id) => id !== clientId
          );
        }
      })

      // **********************************************************************
      // * ensureJobExistsInCache
      .addCase(ensureJobExistsInCache.rejected, (state, action) => {
        const jobId = action.meta.arg;

        // if the job id is not in the error collection, add it
        if (!state.jobsCache.jobIdsWithLoadError.includes(jobId)) {
          state.jobsCache.jobIdsWithLoadError.push(jobId);
        }

        // if the job id is in the pending load collection, remove it
        if (state.jobsCache.jobIdsPendingLoad.includes(jobId)) {
          state.jobsCache.jobIdsPendingLoad = state.jobsCache.jobIdsPendingLoad.filter((id) => id !== jobId);
        }
      });
  }
});

// **********************************************************************
// * actions

export const {
  clientAddedToCache,
  clientLoadingCompleted,
  clientLoadingStarted,
  jobAddedToCache,
  jobLoadingCompleted,
  jobLoadingStarted
} = lookupsSlice.actions;

// **********************************************************************
// * selectors

/**
 * Select a client by id.
 * @param {object} state - The redux state object.
 * @param {*} clientId - The id of the client to select.
 * @returns The client with the given id.
 */
export const selectClient = (state, clientId) =>
  state.lookups.clientsCache.clients.find((client) => client.id === clientId);

/**
 * Select clients.
 * @param {object} state - The redux state object.
 * @returns The clients in the cache.
 */
export const selectClients = (state) => state.lookups.clientsCache.clients;

/**
 * Select a job by id.
 * @param {object} state - The redux state object.
 * @param {*} jobId - The id of the job to select.
 * @returns The job with the given id.
 */
export const selectJob = (state, jobId) => state.lookups.jobsCache.jobs.find((job) => job.id === jobId);

// **********************************************************************
// * reducer

export default lookupsSlice.reducer;
