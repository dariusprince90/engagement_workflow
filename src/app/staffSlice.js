import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import staffApi from '../api/centralDataStore/staffApi';

// **********************************************************************
// * constants

// **********************************************************************
// * functions

// **********************************************************************
// * thunks

/**
 * ensure a staff exists in our cache for a given staff id or azure aad object id
 *
 * * NOTE: either a staffId or azureAdObjectId can be in the incoming arg object
 * *       if both are present, the staffId will be used
 */
export const ensureStaffExistsInCache = createAsyncThunk(
  'staff/ensureStaffExistsInCache',
  async ({ staffId, azureAdObjectId }, { getState, dispatch, rejectWithValue }) => {
    const state = getState();
    const staffIdentifier = staffId || azureAdObjectId;
    const fields = 'id,preferredFullName,emailAddress,azureAdObjectId';

    // if the staffId or azureAdObjectId is already in the pending load collection or in the load error collection
    // then, do nothing (we don't need to attempt to load the same staff multiple times)
    if (
      state.staff.staffCache.staffIdentifiersPendingLoad.includes(staffIdentifier) ||
      state.staff.staffCache.staffIdentifiersWithLoadError.includes(staffIdentifier)
    ) {
      return;
    }

    // trigger the start of the load
    dispatch(staffLoadingStarted(staffIdentifier));

    // try to get the staff from the cache
    let staff = state.staff.staffCache.staff.find(
      (s) => (!!staffId && s.id === staffId) || (!!azureAdObjectId && s.azureAdObjectId === azureAdObjectId)
    );

    // if the staff was not in the cache
    // then, load the staff from the API and insert it into the cache
    if (!staff) {
      if (!!staffId) {
        staff = await staffApi.getStaff(staffId, fields);
      } else {
        staff = await staffApi.getStaffByAzureAdObjectId(azureAdObjectId, fields);
      }

      // if the staff was not returned from the API
      // then there is an error, return a rejected response
      if (!staff) {
        // trigger the end of the load
        dispatch(staffLoadingCompleted(staffIdentifier));

        // reject
        return rejectWithValue();
      }

      // add the staff to the cache
      dispatch(staffAddedToCache(staff));
    }

    // trigger the end of the load
    dispatch(staffLoadingCompleted(staffIdentifier));
  }
);

// **********************************************************************
// * initial state

const initialState = {
  // info about the authenticated user from MSAL
  userAuthInfo: {
    userName: null,
    displayName: null,
    userObjectId: null
  },

  // cache of staff objects that can be re-used without having to hit the API each time we need the staff
  staffCache: {
    // the collection of staff that has been cached
    staff: [],

    // collection of staff identifiers that are pending load from the API
    staffIdentifiersPendingLoad: [],

    // collection of staff identifiers that had an error when loading
    staffIdentifiersWithLoadError: []
  }
};

// **********************************************************************
// * slice

const staffSlice = createSlice({
  name: 'staff',
  initialState,

  reducers: {
    // when triggered, add the staff to the cache
    staffAddedToCache: (state, action) => {
      const staff = action.payload;
      state.staffCache.staff.push(staff);
    },

    // when triggered, remove the staff identifier from the pending load collection
    staffLoadingCompleted: (state, action) => {
      const staffIdentifierToRemove = action.payload;

      state.staffCache.staffIdentifiersPendingLoad = state.staffCache.staffIdentifiersPendingLoad.filter(
        (staffIdentifier) => staffIdentifier !== staffIdentifierToRemove
      );
    },

    // when triggered, add the staff identifier to the pending load collection (if not already in it)
    staffLoadingStarted: (state, action) => {
      const staffIdentifier = action.payload;

      if (!state.staffCache.staffIdentifiersPendingLoad.includes(staffIdentifier)) {
        state.staffCache.staffIdentifiersPendingLoad.push(staffIdentifier);
      }
    },

    // when triggered, set the user auth info properties
    userAuthInfoChanged: (state, action) => {
      const { userName, displayName, userObjectId } = action.payload;
      state.userAuthInfo = { userName, displayName, userObjectId };
    }
  },

  extraReducers: (builder) => {
    builder.addCase(ensureStaffExistsInCache.rejected, (state, action) => {
      const { staffId, azureAdObjectId } = action.meta.arg;
      const staffIdentifier = staffId || azureAdObjectId;

      // if the staff identifier is not in the error collection, add it
      if (!state.staffCache.staffIdentifiersWithLoadError.includes(staffIdentifier)) {
        state.staffCache.staffIdentifiersWithLoadError.push(staffIdentifier);
      }

      // if the staff identifier is in the pending load collection, remove it
      if (state.staffCache.staffIdentifiersPendingLoad.includes(staffIdentifier)) {
        state.staffCache.staffIdentifiersPendingLoad = state.staffCache.staffIdentifiersPendingLoad.filter(
          (si) => si !== staffIdentifier
        );
      }
    });
  }
});

// **********************************************************************
// * actions

export const { staffAddedToCache, staffLoadingCompleted, staffLoadingStarted, userAuthInfoChanged } =
  staffSlice.actions;

// **********************************************************************
// * selectors

// selects the current users' staff object from the cache
export const selectCurrentUserFromCache = (state) => {
  const userObjectId = state.staff.userAuthInfo.userObjectId;
  const staff = state.staff.staffCache.staff.find((s) => s.azureAdObjectId === userObjectId);

  return staff;
};

// select a staff by id
export const selectStaff = (state, staffId) => state.staff.staffCache.staff.find((s) => s.id === staffId);

// selects the user's auth info
export const selectUserAuthInfo = (state) => state.staff.userAuthInfo;

// **********************************************************************
// * reducer

export default staffSlice.reducer;
