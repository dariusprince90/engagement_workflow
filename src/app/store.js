import { configureStore } from '@reduxjs/toolkit';

import appLayoutReducer from '../components/common/appLayout/appLayoutSlice';
import headerReducer from '../components/common/header/headerSlice';
import lookupsReducer from './lookupsSlice';
import neiFileAttachmentModalReducer from '../components/screens/newEngagementInstance/components/fileAttachmentModal/fileAttachmentModalSlice';
import newEngagementInstanceReducer from '../components/screens/newEngagementInstance/newEngagementInstanceSlice';
import sideBarReducer from '../components/common/sideBar/sideBarSlice';
import staffReducer from './staffSlice';

const options = {
  reducer: {
    appLayout: appLayoutReducer,
    lookups: lookupsReducer,
    neiFileAttachmentModal: neiFileAttachmentModalReducer,
    newEngagementInstance: newEngagementInstanceReducer,
    pageHeader: headerReducer,
    staff: staffReducer,
    sideBar: sideBarReducer
  }
};

export const store = configureStore(options);
