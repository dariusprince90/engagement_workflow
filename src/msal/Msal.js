import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { MsalProvider, MsalAuthenticationTemplate } from '@azure/msal-react';
import { EventType, InteractionType, PublicClientApplication } from '@azure/msal-browser';
import { useDispatch } from 'react-redux';

import * as config from '../configs/config';
import { userAuthInfoChanged } from '../app/staffSlice';

import Error from './Error';
import MsalAppInsights from './MsalAppInsights';
import PageOverlay from '../components/common/pageOverlay/PageOverlay';

const msalConfig = {
  auth: {
    clientId: config.MSAL.webApp.clientId,
    authority: config.MSAL.webApp.authority,
    redirectUri: window.location.origin
  }
};

export const msalInstance = new PublicClientApplication(msalConfig);

const propTypes = {
  children: PropTypes.element.isRequired
};

const Msal = ({ children }) => {
  const dispatch = useDispatch();

  // once authenticated, set info about the current user into state
  useEffect(() => {
    const callbackId = msalInstance.addEventCallback((message) => {
      // only process specific event types
      if (message.eventType === EventType.HANDLE_REDIRECT_END) {
        const accounts = msalInstance.getAllAccounts();

        if (accounts.length > 0) {
          const account = accounts[0];
          const userName = account.username.toLowerCase();
          const displayName = account.name;
          const userObjectId = account.idTokenClaims.oid;

          dispatch(userAuthInfoChanged({ userName, displayName, userObjectId }));
        }
      }
    });

    return () => {
      if (callbackId) {
        msalInstance.removeEventCallback(callbackId);
      }
    };
  }, [dispatch]);

  return (
    <MsalProvider instance={msalInstance}>
      <MsalAppInsights>
        <MsalAuthenticationTemplate
          interactionType={InteractionType.Redirect}
          errorComponent={Error}
          loadingComponent={() => <PageOverlay isAppLoader />}>
          {children}
        </MsalAuthenticationTemplate>
      </MsalAppInsights>
    </MsalProvider>
  );
};

Msal.propTypes = propTypes;

export default Msal;
