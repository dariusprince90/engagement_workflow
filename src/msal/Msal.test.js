/* eslint-disable react/prop-types */

import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { act, render } from '@testing-library/react';
import faker from '@faker-js/faker';

import * as msalBrowser from '@azure/msal-browser';
import * as reactRedux from 'react-redux';

import Msal, { msalInstance } from './Msal';
import * as staffSlice from '../app/staffSlice';

// **********************************************************************
// * constants

const componentContents = faker.lorem.sentence();
const componentContentsForSnapshot = 'This is the contents for the snapshot.';

const FakeComponent = () => <fake-component>{componentContents}</fake-component>;
const FakeComponentForSnapshot = () => <fake-component>{componentContentsForSnapshot}</fake-component>;

const mockDispatch = jest.fn();

const componentToRender = (
  <Msal>
    <FakeComponent />
  </Msal>
);

const componentToRenderForSnapshot = (
  <Msal>
    <FakeComponentForSnapshot />
  </Msal>
);

let eventCallbacks = [];
const invokeAllCallbacks = (eventMessage) => {
  eventCallbacks.forEach((callback) => {
    callback(eventMessage);
  });
};

// **********************************************************************
// * mock external dependencies

jest.mock('@azure/msal-react', () => {
  return {
    MsalProvider: ({ children }) => {
      return <fake-msal-provider>{children}</fake-msal-provider>;
    },

    MsalAuthenticationTemplate: ({ children, interactionType, errorComponent, loadingComponent }) => {
      return (
        <fake-msal-auth-template data-interaction-type={interactionType}>
          <error-component>{errorComponent()}</error-component>
          <loading-component>{loadingComponent()}</loading-component>
          {children}
        </fake-msal-auth-template>
      );
    }
  };
});

jest.mock('@azure/msal-browser', () => {
  const originalModule = jest.requireActual('@azure/msal-browser');

  return {
    __esModule: true,
    ...originalModule,

    PublicClientApplication: jest.fn().mockImplementation(() => {
      return {
        addEventCallback: jest.fn(),
        getAllAccounts: jest.fn(),
        removeEventCallback: jest.fn()
      };
    })
  };
});

jest.mock('react-redux');

jest.mock('./MsalAppInsights', () => {
  return {
    __esModule: true,
    default: ({ children }) => {
      return <fake-msal-app-insights>{children}</fake-msal-app-insights>;
    }
  };
});

jest.mock('./Error', () => {
  return {
    __esModule: true,
    default: () => {
      return <fake-error-component />;
    }
  };
});

jest.mock('../components/common/pageOverlay/PageOverlay', () => {
  return {
    __esModule: true,
    default: ({ isAppLoader }) => {
      return <fake-page-overly data-is-app-loader={isAppLoader} />;
    }
  };
});

jest.mock('../app/staffSlice');

// **********************************************************************
// * unit tests

describe('Msal', () => {
  // **********************************************************************
  // * setup

  beforeEach(() => {
    let eventId = 0;
    eventCallbacks = [];

    reactRedux.useDispatch.mockReturnValue(mockDispatch);

    jest.spyOn(msalInstance, 'removeEventCallback');
    jest.spyOn(staffSlice, 'userAuthInfoChanged');

    jest.spyOn(msalInstance, 'addEventCallback').mockImplementation((callbackFn) => {
      eventCallbacks.push(callbackFn);
      eventId += 1;
      return eventId.toString();
    });
  });

  // **********************************************************************
  // * tear-down

  afterEach(() => {
    jest.restoreAllMocks();
    jest.clearAllMocks();
  });

  // **********************************************************************
  // * execution

  describe('rendering', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(componentToRender, div);
    });

    it('renders components nested in oder: provider > app-insights > auth-template > fake-component', () => {
      const tree = renderer.create(componentToRenderForSnapshot).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('functional', () => {
    it('invokes addEventCallback when mounting', () => {
      render(componentToRender);
      expect(msalInstance.addEventCallback).toHaveBeenCalled();
    });

    it('invokes removeEventCallback when unmounting and a callback id exists', () => {
      // * ARRANGE
      msalInstance.addEventCallback.mockReturnValue(1);

      // * ACT
      let { unmount } = render(componentToRender);
      unmount();

      // * ASSERT
      expect(msalInstance.removeEventCallback).toHaveBeenCalled();
    });

    it('does not invoke removeEventCallback when unmounting and no callback id exists', () => {
      // * ARRANGE
      msalInstance.addEventCallback.mockReturnValue(null);

      // * ACT
      let { unmount } = render(componentToRender);
      unmount();

      // * ASSERT
      expect(msalInstance.removeEventCallback).not.toHaveBeenCalled();
    });

    it('dispatches userAuthInfoChanged after HANDLE_REDIRECT_END and the msal instance has at least 1 account', () => {
      // * ARRANGE
      const eventType = msalBrowser.EventType.HANDLE_REDIRECT_END;
      const userName = `${faker.name.firstName()} ${faker.name.lastName()}`;
      const accountName = faker.internet.email();
      const userObjectId = faker.datatype.uuid();
      const eventMessage = { eventType, payload: { account: { username: userName } } };
      const account = { username: userName, name: accountName, idTokenClaims: { oid: userObjectId } };
      const accounts = [account];
      msalInstance.getAllAccounts.mockReturnValue(accounts);

      // * ACT
      render(componentToRender);
      act(() => invokeAllCallbacks(eventMessage));

      // * ASSERT
      expect(mockDispatch).toHaveBeenCalledWith(staffSlice.userAuthInfoChanged());
      expect(staffSlice.userAuthInfoChanged).toHaveBeenCalledWith({
        userName: userName.toLowerCase(),
        displayName: accountName,
        userObjectId
      });
    });

    it('does not dispatch userAuthInfoChanged after HANDLE_REDIRECT_END and the msal instance has no accounts', () => {
      // * ARRANGE
      const eventType = msalBrowser.EventType.HANDLE_REDIRECT_END;
      const eventMessage = { eventType };
      const accounts = [];
      msalInstance.getAllAccounts.mockReturnValue(accounts);

      // * ACT
      render(componentToRender);
      act(() => invokeAllCallbacks(eventMessage));

      // * ASSERT
      expect(mockDispatch).not.toHaveBeenCalled();
      expect(staffSlice.userAuthInfoChanged).not.toHaveBeenCalled();
    });

    it('does not dispatch userAuthInfoChanged after an event that is not being watched', () => {
      // * ARRANGE
      const eventType = faker.random.word();
      const eventMessage = { eventType };

      // * ACT
      render(componentToRender);
      act(() => invokeAllCallbacks(eventMessage));

      // * ASSERT
      expect(mockDispatch).not.toHaveBeenCalled();
      expect(staffSlice.userAuthInfoChanged).not.toHaveBeenCalled();
    });
  });
});
