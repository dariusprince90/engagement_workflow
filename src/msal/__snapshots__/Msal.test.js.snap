// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Msal rendering renders components nested in oder: provider > app-insights > auth-template > fake-component 1`] = `
<fake-msal-provider>
  <fake-msal-app-insights>
    <fake-msal-auth-template
      data-interaction-type="redirect"
    >
      <error-component>
        <fake-error-component />
      </error-component>
      <loading-component>
        <fake-page-overly
          data-is-app-loader={true}
        />
      </loading-component>
      <fake-component>
        This is the contents for the snapshot.
      </fake-component>
    </fake-msal-auth-template>
  </fake-msal-app-insights>
</fake-msal-provider>
`;
